lib/services/flowdock.ts

```diff
@@ -15,129 +15,137 @@ limitations under the License.
-                    // ...but this respects typing
-                    } else {
-                        reject(new Error('Something weird happened in the flowdock SDK'));
-                    }
-                }
-            );
+        // Attempt to find a username by id
+        return this.fetchFromSession(`/organizations/${org}/users/${userId}`)
+        .then((user) => {
+            returnValue.sourceIds.user = user.nick;
+            return returnValue;
         });
     }
 
     /**
-     * Retrieve the private message history with a user
-     * @param event details of the event to consider
-     * @param filter optional criteria that must be met
+     * Promise to turn the generic message format into a specific form to be emitted
+     * @param data - Generic message format object to be encoded
+     * @returns {Bluebird<FlowdockEmitContext>} - Promise that resolves to the emit suitable form
      */
-    public fetchPrivateMessages(event: ReceiptContext, filter: RegExp): Promise<string[]> {
-        // Check that the event being asked about orginated with us
-        if (event.source !== this.serviceName) {
-            return Promise.reject(new Error('Cannot get flowdock thread from non-flowdock event'));
-        }
-        return new Promise<string[]>((resolve, reject) => {
-            // Create an options array for shortlisting, if possible
-            const keywords = filter.source.match(/(\w+)/g);
-            const options = keywords === null ? {} : { search: keywords.join(' ') };
-            // Query the API via the SDK
-            FlowdockService.session.get(
-            `/private/${event.sourceIds.user}/messages`, options,
-            (error?: Error, result?: FlowdockMessage[]) => {
-                // Massage the callback response into promises
-                if (error) {
-                    reject(error);
-                } else if (result) {
-                    resolve(_
-                        // Clean the response to just the content
-                        .map(result, (value: FlowdockMessage) => {
-                            return value.content;
-                        })
-                        // Filter the response to just matches
-                        .filter((value: string) => {
-                            const match = value.match(filter);
-                            return match !== null && match.length > 0;
-                        })
-                    );
-                // This should never happen, since the callback pattern should have error or response
-                // ...but this respects typing
-                } else {
-                    reject(new Error('Something weird happened in the flowdock SDK'));
-                }
+    public makeSpecific(data: TransmitContext): Promise<FlowdockEmitContext> {
+        // Build a string for the title, if appropriate.
+        const titleText = data.first && data.title ? data.title + '\n--\n' : '';
```

Comment:
--- with an empty line above means horizontal rule, ie the stuff below it is a footer
-- with a content line above means subtitle, ie the stuff above is a title